tests/conftest.py

```diff
@@ -86,3 +87,12 @@ def datasets():
         examples.load_airplane(),  # PolyData
         examples.load_structured(),  # StructuredGrid
     ]
+
+
+@fixture()
+def check_vtk_error(caplog):
+    """Ensure that no VTK error is thrown."""
+    yield
+    for record in caplog.get_records("call"):
+        if record.levelno == logging.ERROR and "vtk" in record.message:
```

Comment:
I added this check for `"vtk" in record.message` to try and isolate this tear-down check to just vtk errors.  This is the most straightforward procedure I could figure out, but it will also trip if any other errors are logged with `vtk` in the message.

Other options I can think of:
1) Make this a fixture to capture _any_ logged error
2) I don't know if we can use the Observer functionality directly from https://github.com/pyvista/pyvista/blob/aefb4df72450bedab5078d13d71978b61be2f915/pyvista/utilities/errors.py#L24-L25

Reply:
Observer _should_ work. However it doesn't work in this case! Here is some very odd code demonstrating the observer's ability to catch one error that `vtkStreamTracer` throws, but not the one relevant to this PR. Hmm...

```
import pyvista as pv
import pyvista._vtk as vtk
import numpy as np

x, y, z = np.meshgrid(
    np.arange(-10, 10, 0.5), np.arange(-10, 10, 0.5), np.arange(-10, 10, 0.5)
)
mesh = pv.StructuredGrid(x, y, z)
x2, y2, z2 = np.meshgrid(
    np.arange(-1, 1, 0.5), np.arange(-1, 1, 0.5), np.arange(-1, 1, 0.5)
)
mesh2 = pv.StructuredGrid(x2, y2, z2)

# throws an error if no input data is set
alg = vtk.vtkStreamTracer()
obs = pv.Observer()
obs.observe(alg)
alg.Update()
print(obs.has_event_occurred())
print(obs.get_message())

# does NOT throw an error here, even though one gets printed to stdout
alg = vtk.vtkStreamTracer()
obs = pv.Observer()
obs.observe(alg)
alg.SetInputDataObject(mesh)
alg.SetSourceData(mesh2)
alg.Update()
print(obs.has_event_occurred())
print(obs.get_message())
```

Reply:
I should point out that even if the observer did work correctly, it is very difficult to add an observer onto the `alg` inside the actual `streamlines_from_source` method.

Reply:
I should point out that even if the observer did work correctly, it is very difficult to add an observer onto the `alg` inside the actual `streamlines_from_source` method.

Reply:
> Here is some very odd code demonstrating the observer's ability to catch one error that vtkStreamTracer throws, but not the one relevant to this PR. Hmm...

I think it doesn't catch the error because the error is thrown in 
`vtkStreamingDemandDrive` or something like this.  See the error message above.

Reply:
Here is the output of that snippet.

The first error comes from `vtkDemandDrivenPipeline` and is detected by the observer, while the second error comes from `vtkStreamingDemandDrive`

```
True
Input port 0 of algorithm vtkStreamTracer(0x556a7c30f150) has 0 connections but is not optional.
False
None
2021-06-13 08:05:50.343 (   0.096s) [         1555740]vtkDemandDrivenPipeline:666    ERR| vtkCompositeDataPipeline (0x556a7c347630): Input port 0 of algorithm vtkStreamTracer(0x556a7c30f150) has 0 connections but is not optional.
ERROR:root:Input port 0 of algorithm vtkStreamTracer(0x556a7c30f150) has 0 connections but is not optional.
2021-06-13 08:05:50.344 (   0.096s) [         1555740]vtkStreamingDemandDrive:871    ERR| vtkStreamingDemandDrivenPipeline (0x556a7c348930): The update extent specified in the information for output port 0 on algorithm vtkTrivialProducer(0x556a7c348850) is 0 39 0 39 0 39, which is outside the whole extent 0 3 0 3 0 3.
ERROR:root:The update extent specified in the information for output port 0 on algorithm vtkTrivialProducer(0x556a7c348850) is 0 39 0 39 0 39, which is outside the whole extent 0 3 0 3 0 3.
```